code/game/cas_manager/datums/cas_fire_envelope.dm

```diff
@@ -273,15 +266,85 @@
 			SPAN_HIGHDANGER("YOU HEAR SOMETHING FLYING CLOSER TO YOU!") , SHOW_MESSAGE_AUDIBLE \
 		)
 
-	sleep(flyto_period)
+/datum/cas_fire_envelope/proc/second_chat_warning(atom/initial_turf)
+	var/relative_dir
+	for(var/mob/mob in range(15, initial_turf))
+		if(get_turf(mob) == initial_turf)
+			relative_dir = 0
+		else
+			relative_dir = Get_Compass_Dir(mob, initial_turf)
+
+		var/ds_identifier = "LARGE BIRD"
+		if (mob.mob_flags & KNOWS_TECHNOLOGY)
+			ds_identifier = "DROPSHIP"
+
+		mob.show_message( \
+			SPAN_HIGHDANGER("A [ds_identifier] FLIES [SPAN_UNDERLINE(relative_dir ? uppertext(("TO YOUR " + dir2text(relative_dir))) : uppertext("right above you"))]!"), SHOW_MESSAGE_VISIBLE, \
+			SPAN_HIGHDANGER("YOU HEAR SOMETHING GO [SPAN_UNDERLINE(relative_dir ? uppertext(("TO YOUR " + dir2text(relative_dir))) : uppertext("right above you"))]!"), SHOW_MESSAGE_AUDIBLE \
+		)
+
+/datum/cas_fire_envelope/proc/third_chat_warning(atom/initial_turf)
+	var/relative_dir
+	for(var/mob/mob in range(10, initial_turf))
+		if(get_turf(mob) == initial_turf)
+			relative_dir = 0
+		else
+			relative_dir = Get_Compass_Dir(mob, initial_turf)
+
+		var/ds_identifier = "LARGE BIRD"
+		if (mob.mob_flags & KNOWS_TECHNOLOGY)
+			ds_identifier = "DROPSHIP"
+
+		mob.show_message( \
+			SPAN_HIGHDANGER("A [ds_identifier] FIRES [SPAN_UNDERLINE(relative_dir ? uppertext(("TO YOUR " + dir2text(relative_dir))) : uppertext("right above you"))]!"), 1, \
+			SPAN_HIGHDANGER("YOU HEAR SOMETHING FIRE [SPAN_UNDERLINE(relative_dir ? uppertext(("TO YOUR " + dir2text(relative_dir))) : uppertext("right above you"))]!"), 2 \
+		)
```

Comment:
`first_chat_warning`, `second_chat_warning` and `third_chat_warning` can be refactored to one proc where you give it two arguments: range (e.g. 15 or 10) and precise (whether you get directions in the messages)